prusti-contracts/prusti-contracts/src/core_spec/default.rs

```diff
@@ -0,0 +1,45 @@
+use crate::*;
+
+#[extern_spec]
+trait Default {
+    #[refine_spec(where Self: Copy + PureDefault, [pure])]
+    fn default() -> Self;
+}
+
+/// Specifies that `Default::default`, if implemented, is a pure method, allowing its usage in specs.
+///
+/// Does not apply to types that do not implement `Copy`, since pure methods can only involve `Copy` types.
+pub auto trait PureDefault {}
+
+// analogous to https://github.com/rust-lang/rust/blob/872631d0f0fadffe3220ab1bd9c8f1f2342341e2/library/core/src/default.rs#L190-L202
+macro_rules! default_spec {
+    ($t:ty, $v:expr) => {
+        #[extern_spec]
+        impl Default for $t {
+            #[pure]
+            #[ensures(result == $v)]
+            fn default() -> Self;
+        }
+    };
+}
+
+default_spec! { (), () }
```

Comment:
Can we also specify the `Default` for non-unit tuples? [It exists for up to 12-item tuples.](https://doc.rust-lang.org/stable/std/primitive.tuple.html#impl-Default-for-(T%2C))